internals/overlord/servstate/manager.go

```diff
@@ -550,19 +346,15 @@ func (m *ServiceManager) SetServiceArgs(serviceArgs map[string][]string) error {
 // exits), it would generate a runtime error as the reaper would already be dead.
 // This function returns a slice of service names to stop, in dependency order.
 func servicesToStop(m *ServiceManager) ([]string, error) {
-	mgrPlan, err := m.Plan()
-	if err != nil {
-		return nil, err
-	}
-
+	planSnapshot := m.plan()
 	// Get all service names in plan.
-	services := make([]string, 0, len(mgrPlan.Services))
-	for name := range mgrPlan.Services {
+	var services []string
```

Comment:
Why the change here? The `make(T, 0, cap)` idiom pre-allocates a slice with the correct capacity, avoiding log(N) allocates in the append loop. Not that low-level performance is critical here, but doesn't hurt to leave it.

Reply:
Sorry about this Ben. I did not even recall I did this, so not something I would normally have changed :( This happened during a rebase operation, and I think I was just a little bit too hasty, and did not pay attention to the conflict I was trying to resolve. I should take more care!

Reply:
Fixed!

Reply:
Resolving.

Reply:
No problem, and no need to apologize! You're one of the most careful programmers I've seen. :-)